instance/models/openedx_instance.py

```diff
@@ -52,73 +62,105 @@ class OpenEdXInstance(Instance, OpenEdXAppConfiguration, OpenEdXDatabaseMixin, O
     """
 
     # Most settings/fields are inherited from OpenEdXAppConfiguration
-    sub_domain = models.CharField(max_length=50)
-    base_domain = models.CharField(max_length=50, blank=True)
+
+    # Internal domains are controlled by us and their DNS records are automatically set to point to the current active
+    # appserver. They are generated from a unique prefix (given as 'sub_domain' in instance factories) and the value of
+    # DEFAULT_INSTANCE_BASE_DOMAIN at instance creation time. They cannot be blank and are normally never changed after
+    # the instance is created.
+    # External domains on the other hand are controlled by the customer and are optional. We use external domains in
+    # preference to internal domains when displaying links to the instance in the UI and when passing domain related
+    # settings to ansible vars when provisioning appservers.
+    # The `domain`, `lms_preview_domain`, and `studio_domain` methods are useful when you need to access the external
+    # domain if present, otherwise fall back to the corresponding internal domain.
```

Comment:
@mtyaka "The `domain`, `lms_preview_domain`, and `studio_domain` properties below are useful if you need to access corresponding domains regardless of whether an instance uses external domains or not (they return the external domain if set, and fall back to the corresponding internal domain otherwise)."

Thanks for adding a detailed comment here!